specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2022-06-01-privatepreview/mysql.json

```diff
@@ -0,0 +1,2964 @@
+      "post": {
+        "tags": [
+          "GetPrivateDnsZoneSuffix"
+        ],
+        "operationId": "GetPrivateDnsZoneSuffix_Execute",
+        "x-ms-examples": {
+          "GetPrivateDnsZoneSuffix": {
+            "$ref": "./examples/GetPrivateDnsZoneSuffix.json"
+          }
+        },
+        "description": "Get private DNS zone suffix in the cloud.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/GetPrivateDnsZoneSuffixResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMySQL/assessForMigration": {
+      "post": {
+        "tags": [
+          "AssessForMigration"
+        ],
+        "operationId": "AssessForMigration_Assess",
+        "x-ms-examples": {
+          "PerformMigrationAssessment": {
+            "$ref": "./examples/AssessForMigration.json"
+          }
+        },
+        "description": "Performs Migration Assessment for a single server",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "MigrationAssessmentRequest",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/MigrationAssessmentRequest"
+            },
+            "description": "The required parameters for performing a migration assessment on a single server"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/MigrationAssessmentResource"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMySQL/flexibleServers/{serverName}/cutoverMigration": {
```

Comment:
>cutoverMigration [](http://example.com/codeflow?start=132&length=16)

POST actions should have verb names. Any reason this action isn't just named "migrate"?

Reply:
Migrate would mean "start data migration".
Here cutoverMigration means complete the migration that was started

Reply:
OK. So this PR only adds two POST actions: assessForMigration and cutoverMigration. Presumably assessForMigration isn't starting a migration, so how does a client start a migration?

Reply:
hi

Reply:
This comment and all following comments are on the behalf of @pranavi-dandu 

The start migration is part of the server create call. We added a new enum string in CreateMode property for it. When the create mode is "Migrate", the server creation and data migration take place

Reply:
OK, I thought that could be the case. I see flexible servers don't have provisioningState today, which would help with ARM consistency in this scenario. Presumably, there is some state variable that indicates the migration has started and later completed, but the CreateMode approach seems to mix request and state a little bit. Given the history and that this is an incremental change, the only additional feedback I have on this API is that I'd recommend using completeMigration as the POST action to indicate it only does something if a migration is in progress.

Reply:
The reason cutoverMigration is less clear is because it implies you could do the whole job with that one action. It doesn't give any clue that some other action has to be done first. Using completeMigration for the name, clearly indicates that a start action has to be done first. This is my last comment in this thread: I won't block ARM approval on changing the name: you can decide.